File: solutions/r/hamming/1/hamming.R
# This is a stub function to take two strings
# and calculate the hamming distance
hamming <- function(strand1, strand2) {
  if (nchar(strand1) == nchar(strand2)){
    diff = 0
    for (i in 1:nchar(strand1)){
      if (substring(strand1, i, i) != substring(strand2, i, i)){
        diff <- diff + 1
      }
    }
  diff
  } else {
  stop()
  } 
}

File: solutions/c/hamming/7/hamming.c
#include "hamming.h"

int compute(const char *lhs, const char *rhs) {
  if (!lhs || !rhs)
    return -1;

  int distance = 0;
  for (; *lhs && *rhs; lhs++, rhs++) {
    if (*lhs != *rhs) {
      distance++;
    }
  }
  return (*lhs || *rhs) ? -1 : distance;
}
